src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/DataContractJsonSerializer.cs

```diff
@@ -694,41 +687,41 @@ public override void WriteEndObject(XmlDictionaryWriter writer)
         }
 
 
-        public override void WriteObject(Stream stream, object graph)
+        public override void WriteObject(Stream stream, object? graph)
         {
             CheckNull(stream, nameof(stream));
             XmlDictionaryWriter jsonWriter = JsonReaderWriterFactory.CreateJsonWriter(stream, Encoding.UTF8, false); //  ownsStream
             WriteObject(jsonWriter, graph);
             jsonWriter.Flush();
         }
 
-        public override void WriteObject(XmlWriter writer, object graph)
+        public override void WriteObject(XmlWriter writer, object? graph)
         {
             WriteObjectHandleExceptions(new JsonWriterDelegator(writer, this.DateTimeFormat), graph);
         }
 
-        public override void WriteObject(XmlDictionaryWriter writer, object graph)
+        public override void WriteObject(XmlDictionaryWriter writer, object? graph)
         {
             WriteObjectHandleExceptions(new JsonWriterDelegator(writer, this.DateTimeFormat), graph);
         }
 
-        public override void WriteObjectContent(XmlWriter writer, object graph)
+        public override void WriteObjectContent(XmlWriter writer, object? graph)
         {
             WriteObjectContentHandleExceptions(new JsonWriterDelegator(writer, this.DateTimeFormat), graph);
         }
 
-        public override void WriteObjectContent(XmlDictionaryWriter writer, object graph)
+        public override void WriteObjectContent(XmlDictionaryWriter writer, object? graph)
         {
             WriteObjectContentHandleExceptions(new JsonWriterDelegator(writer, this.DateTimeFormat), graph);
         }
 
-        public override void WriteStartObject(XmlWriter writer, object graph)
+        public override void WriteStartObject(XmlWriter writer, object? graph)
         {
             // No need to pass in DateTimeFormat to JsonWriterDelegator: no DateTimes will be written in start object
             WriteStartObjectHandleExceptions(new JsonWriterDelegator(writer), graph);
         }
 
-        public override void WriteStartObject(XmlDictionaryWriter writer, object graph)
+        public override void WriteStartObject(XmlDictionaryWriter writer, object? graph)
```

Comment:
not sure if affects public APIs but `internal static bool CheckIfJsonNameRequiresMapping(string jsonName)` should take nullable jsonName

Reply:
It doesn't affect public API since it is only called in 2 places that already check for `null`. I added the `?`.